### PR TITLE
Set repo priority to 'undef'

### PR DESCRIPTION
**Pull request by chuman**
This should be set to undef for:
  if $priority { ... }
to be ignored in repo/yum.pp

### CODE REVIEW

**Comment by skpy** [comment]
The [yumrepo priority](http://docs.puppetlabs.com/references/3.4.stable/type.html#yumrepo-attribute-priority) parameter accepts `absent` as a valid value, which will ensure that no priority line is present in the generated repo file.

Locally, I've added the following line to the [yum repo spec](https://github.com/covermymeds/puppet-gluster/blob/master/spec/classes/repo_yum_spec.rb#L43) test:
```
should_not create_package('yum-plugin-priorities')
```
And `rake spec` runs the tests successfully.  This indicates to me that Puppet is handling `absent` similar to `undef`.

**However**, I've also run `puppet apply` with the following trivial manifest:
```
$foo = absent
if $foo {
  notice("FOO IS TRUE")
}
```
and received the following output:
```
Notice: Scope(Class[main]): FOO IS TRUE
Notice: Compiled catalog for vagrant-smerrill.dev in environment production in 0.02 seconds
Notice: Finished catalog run in 0.05 seconds
```
So Puppet appears to be treating the bare word `absent` as boolean true within the conditional evaluation.

TL;DR: `absent` may well be installing the `yum-plugins-priorities` package even though no priority line will be defined within the generated repo file.  It's not clear to me why the spec test claim it's not being installed; but this PR is a better solution all around.

Thanks!
